components/badge/badge_input.c

```diff
@@ -76,6 +77,7 @@ badge_input_add_event(uint32_t button_id, bool pressed, bool in_isr)
 				ets_printf("badge_input: input queue full.\n");
 			}
 		}
+		badge_vibrator_buzz(70000);
```

Comment:
This method is often called from a different thread than the main thread. This needs a mutex to avoid doing haptic feedback during a normal badge_vibrator_activate() call.